src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -40,17 +40,19 @@ public Swerve() {
         gyro = new Pigeon2(Constants.Swerve.pigeonID);
         gyro.configFactoryDefault();
         zeroGyro();
-
-        Timer.delay(1.0);
-        resetModulesToAbsolute();
         
-        swerveOdometry = new SwerveDriveOdometry(Constants.Swerve.swerveKinematics, getYaw(), getModulePositions());
         mSwerveMods = new SwerveModule[] {
             new SwerveModule(0, Constants.Swerve.Mod0.constants),
             new SwerveModule(1, Constants.Swerve.Mod1.constants),
             new SwerveModule(2, Constants.Swerve.Mod2.constants),
             new SwerveModule(3, Constants.Swerve.Mod3.constants)
         };
+
+        
+        Timer.delay(1.0);
```

Comment:
Be *very* cautious about adding timer delays in the robot code.  Keep in mind that various parts of the code is called every 0.02 seconds, so if you put in a delay of a second, it might end up pausing the whole robot for 1 second and throwing off timing/scheduling of other things.

Could a much smaller delay be used here instead?

What is the reason for having this delay?

Reply:
The delay is just to let all the swerve modules initialize and get configured and then snap them to the zero position, so it could probably be much shorter.

Reply:
Good explanation.  Then the timing isn't as important as when and where instances of the Swerve class are created.  If the location is safe for a long running task, then leaving it at 1.0 seconds is fine.  You don't want to cut the time down and risk failing because they aren't fully initialized.

Reply:
You could also check the SwerveModule class and see if there is method that tells you if the module is ready to use.  It may also be the case that the SwerveModule constructor does not return until it is fully ready, in which case the delay is completely unnecessary.